tests/orm/data/test_code.py

```diff
@@ -0,0 +1,48 @@
+"""Tests for the `Code` class."""
+# pylint: disable=redefined-outer-name
+import pytest
+
+from aiida.orm import Code
+
+
+@pytest.fixture
+def create_codes(tmpdir, aiida_localhost):
+    """Create a local and remote `Code` instance."""
+    import os
+
+    filename = 'add.sh'
+    filepath = os.path.join(tmpdir, filename)
+
+    with open(filepath, 'w'):
+        pass
+
+    code_local = Code(local_executable=filename, files=[filepath]).store()
```

Comment:
Nitpick: So, what is the purpose of the parameter `files` when initializing codes? It seems a bit strange to me having to manually add the executable there. If this could have been any other file then maybe it would be less confusing to just make two files for this? (or maybe also use `bin/true` as a local code).

Reply:
This is due to the nature of "local" codes. Remote codes point to a binary that is already preinstalled on the target computer. Local codes, in contrast, are "codes" or "scripts" you have on the machine where AiiDA is running and you want to run on a target machine. This might include multiple files: think some Python package or modules and a single script that actually runs the calculation. When running a "local" code, AiiDA will first upload all these files to the remote machine and then call the script. This is the interface as it was coded at the time and it requires all files to be passed in `files` and it should explicitly tell which one is the "executable"

Reply:
Ah, ok, so it is not a "local code" in the sense of a code you execute locally, but just in the sense that the executable has to be copied. So can a code be initialized with both `local_executable' AND `remote_computer_exec`? Does it make any sense to specify `files` when using `remote_computer_exec`?

Reply:
> Ah, ok, so it is not a "local code" in the sense of a code you execute locally, but just in the sense that the executable has to be copied.

Exactly, but not just the executable, also any other file that it needs to execute it remotely, think module files etc.

> So can a code be initialized with both `local_executable' AND `remote_computer_exec`? Does it make any sense to specify `files`when using`remote_computer_exec`?

No, these are mutually exclusive.